src/Streamly/Internal/Data/Stream/IsStream/Top.hs

```diff
@@ -356,23 +370,34 @@ leftJoin eq s1 s2 = Stream.evalStateT (return False) $ do
 --
 -- Time: O(m + n)
 --
--- /Unimplemented/
+-- /Pre-release/
 {-# INLINE hashLeftJoin #-}
-hashLeftJoin :: -- Hashable b =>
-    (a -> b -> Bool) -> t m a -> t m b -> t m (a, Maybe b)
-hashLeftJoin = undefined
+hashLeftJoin ::  (IsStream t, Ord a, Monad m, Monad (t m)) =>
+    t m a -> t m a -> t m (a, Maybe a)
```

Comment:
The signature should be similar to hashInnerJoin above i.e. `t m (k, a) -> t m (k, b) -> t m (k, a, Maybe b)`.